File: apps/src/chat/backend/packets/websocket/enter_room_packet.py
from ..packet import Packet

class EnterRoomPacket(Packet):

    def __init__(self, username: str):
        self._username = username

    def to_data(self):
        return {
            "target": self._username,
            "type": "enter_room"
        }

    @staticmethod
    def from_data(data):
        return EnterRoomPacket(
            data["username"]
        )

File: apps/src/chat/frontend/js/packets/websocket/EnterRoomPacket.js
export default class EnterRoomPacket {

    #targetUsername;

    constructor(
        targetUsername
    ){
        this.#targetUsername = targetUsername;
    }

    toData(){
        return {
            target_username: this.#targetUsername,
            type: "enter_room"
        }
    }
}